tests/Integration/Momento.Sdk.Tests/VectorIndexControlTest.cs

```diff
@@ -17,30 +17,43 @@ public async Task CreateListDelete_HappyPath()
-        Assert.True(createResponse is CreateVectorIndexResponse.Success, $"Unexpected response: {createResponse}");
-        
-        var listResponse = await vectorIndexClient.ListIndexesAsync();
-        Assert.True(listResponse is ListVectorIndexesResponse.Success, $"Unexpected response: {listResponse}");
-        var listOk = (ListVectorIndexesResponse.Success)listResponse;
-        Assert.Contains(listOk.IndexNames, name => name == indexName);
-        
-        var deleteResponse = await vectorIndexClient.DeleteIndexesAsync(indexName);
-        Assert.True(deleteResponse is DeleteVectorIndexResponse.Success, $"Unexpected response: {deleteResponse}");
+
+        try
+        {
+            var createResponse = await vectorIndexClient.CreateIndexAsync(indexName, numDimensions);
+            Assert.True(createResponse is CreateVectorIndexResponse.Success, $"Unexpected response: {createResponse}");
+
+            var listResponse = await vectorIndexClient.ListIndexesAsync();
+            Assert.True(listResponse is ListVectorIndexesResponse.Success, $"Unexpected response: {listResponse}");
+            var listOk = (ListVectorIndexesResponse.Success)listResponse;
+            Assert.Contains(listOk.IndexNames, name => name == indexName);
+        }
+        finally
+        {
+            var deleteResponse = await vectorIndexClient.DeleteIndexesAsync(indexName);
+            Assert.True(deleteResponse is DeleteVectorIndexResponse.Success, $"Unexpected response: {deleteResponse}");
+        }
     }
 
     [Fact]
     public async Task CreateIndexAsync_AlreadyExistsError()
     {
         var indexName = $"index-{Utils.NewGuidString()}";
         const int numDimensions = 3;
-        
-        var createResponse = await vectorIndexClient.CreateIndexAsync(indexName, numDimensions);
-        Assert.True(createResponse is CreateVectorIndexResponse.Success, $"Unexpected response: {createResponse}");
-        
-        var createAgainResponse = await vectorIndexClient.CreateIndexAsync(indexName, numDimensions);
-        Assert.True(createAgainResponse is CreateVectorIndexResponse.AlreadyExists, $"Unexpected response: {createAgainResponse}");
+
+        try
+        {
+            var createResponse = await vectorIndexClient.CreateIndexAsync(indexName, numDimensions);
+            Assert.True(createResponse is CreateVectorIndexResponse.Success, $"Unexpected response: {createResponse}");
+
+            var createAgainResponse = await vectorIndexClient.CreateIndexAsync(indexName, numDimensions);
+            Assert.True(createAgainResponse is CreateVectorIndexResponse.AlreadyExists, $"Unexpected response: {createAgainResponse}");
+        }
+        finally
+        {
+            var deleteResponse = await vectorIndexClient.DeleteIndexesAsync(indexName);
+            Assert.True(deleteResponse is DeleteVectorIndexResponse.Success, $"Unexpected response: {deleteResponse}");
```

Comment:
I'm not sure if we should assert that delete is successful. This clean-up finally block is best-effort. IMO we should do
```
finally {
  try {
     var deleteResponse = await vectorIndexClient.DeleteIndexesAsync(indexName);
  } catch (e: Exception) {
    // log and continue;
  }

}

```

Otherwise the failure from this finally block can mask the actual failure from the try block above.

Reply:
agreed. also might be good to fast follow this with a higher-order helper function called `withTestVectorIndex` that encapsulates the cleanup so that we have an easy repeatable pattern for it.

Reply:
I can switch to just throwing an exception. We need to avoid a passing test containing a failing delete call.